src/main/java/com/acme/dbo/txlog/Facade.java

```diff
@@ -1,11 +1,148 @@
 package com.acme.dbo.txlog;
 
+import java.util.Objects;
+
+import static java.lang.String.valueOf;
+
 public class Facade {
+
+    private static final String PRIMITIVE = "primitive";
+    private static final String CHAR = "char";
+    private static final String REFERENCE = "reference";
+    private static final String STRING = "string";
+
+    private static final int OTHER_MESSAGE = 0;
+    private static final int INT_MESSAGE = 1;
+    private static final int BYTE_MESSAGE = 2;
+    private static final int STRING_MESSAGE = 3;
+
+    private static int intSum = 0;
+    private static byte byteSum = 0;
+    private static int stringCount = 0;
+    private static String lastString;
+    private static StringBuilder stringBuilder = new StringBuilder();
+
     public static void log(int message) {
-        System.out.println("primitive: " + message);
+
+        try {
+            intSum = Math.addExact(intSum, message);
+            logMessage(decorateMessage(intSum), intSum);
+        } catch (ArithmeticException e) {
+            logMessage(decorateMessage(intSum), intSum);
+            logMessage(decorateMessage(message), message);
+            intSum = message;
+        }
+        clearByMessageType(INT_MESSAGE);
     }
 
     public static void log(byte message) {
-        System.out.println("primitive: " + message);
+
+        if ((int)message + byteSum >= Byte.MAX_VALUE) {
+            byteSum = 0;
+        } else {
+            byteSum += message;
+        }
+        logMessage(decorateMessage(byteSum), byteSum);
+        logMessage(decorateMessage(message), message);
+        clearByMessageType(BYTE_MESSAGE);
+    }
+
+    public static void log(char message) {
+        logMessage(decorateMessage(message), valueOf(message));
+        clearByMessageType(OTHER_MESSAGE);
+    }
+
+    public static void log(String message) {
+        if (Objects.equals(message, lastString)) {
+            stringCount++;
+        } else {
+            stringCount = 0;
+        }
+        logMessage(decorateMessage(message), message);
+        lastString = message;
+        clearByMessageType(STRING_MESSAGE);
+    }
+
+    public static void log(boolean message) {
+        logMessage(decorateMessage(message), valueOf(message));
+        clearByMessageType(OTHER_MESSAGE);
+    }
+
+    public static void log(Object message) {
+        logMessage(decorateMessage(message), message);
+        clearByMessageType(OTHER_MESSAGE);
+    }
+
+    private static void logMessage(String prefix, Object message) {
+        stringBuilder.append(prefix).append(": ").append(message).append(System.lineSeparator());
+    }
+
+    private static void logMessage(String prefix, String message) {
+        if (stringCount == 0) {
+            stringBuilder.append(prefix).append(": ").append(message).append(System.lineSeparator());
+        } else {
+            String substring = prefix + ": " + message;
```

Comment:
Maybe store string count as field? Just to simplify business logic of string aggregating?